### PR TITLE
GitHub tree

### PR DESCRIPTION
**Pull request by asfaltboy**
# Example query results
<details>
  <summary>Results</summary>
  
```
Add example SQL query results here (please include the input queries as well)
```
</details>

### CODE REVIEW

**Comment by cbruno10** [comment]
Hey @asfaltboy , thanks for opening this PR and sorry we've missed it for so long! I'm going to re-open it and we'll review it early next week. If we have any feedback or suggestions, we'll add them back in the PR.

File: github/table_github_tree.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableGitHubTree(ctx context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "github_tree",
+		Description: "Tree in the given repository, lists files in the git tree",
+		Get: &plugin.GetConfig{
```
Comment (bigdatasourav) [inline github/table_github_tree.go:19]: Could you please replace get with the list as below -
```
List: &plugin.ListConfig{
			Hydrate:           tableGitHubTreeGet,
			ShouldIgnoreError: isNotFoundError([]string{"404"}),
			KeyColumns: []*plugin.KeyColumn{
				{Name: "repository_full_name", Require: plugin.Required},
				{Name: "tree_sha", Require: plugin.Required},
				{Name: "name", Require: plugin.Optional},
			},
		},
```

File: github/table_github_tree.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+}
+
+//// GET FUNCTION
+
+func tableGitHubTreeGet(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	logger.Trace("Connecting to client")
+	client := connect(ctx, d)
+
+	if h.Item != nil {
```
Comment (bigdatasourav) [inline github/table_github_tree.go:46]: Please remove this if-else block for hydrate data, as it is not required here.

File: github/table_github_tree.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+		item := h.Item
+		logger.Trace("Got hydrate data", item)
+	} else {
+		logger.Trace("No hydrate data")
+	}
+	quals := d.KeyColumnQuals
+	logger.Trace("Parsing key column quals", quals)
+	fullName := quals["repository_full_name"].GetStringValue()
+	sha := quals["tree_sha"].GetStringValue()
```
Comment (bigdatasourav) [inline github/table_github_tree.go:55]: Add a check for empty values like below -
```
// check if fullName and sha is empty
	if fullName == "" || sha == "" {
		return nil, nil
	}
```

File: github/table_github_tree.go [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+	if err != nil {
+		logger.Error("Error getting tree", err)
+		return nil, err
+	}
+
+	getResp := getResponse.(GetResponse)
+	tree := getResp.tree
+	if tree != nil {
+		logger.Trace("Returning tree", tree)
+		return tree, nil
```
Comment (bigdatasourav) [inline github/table_github_tree.go:83]:
```suggestion
		d.StreamListItem(ctx, tree)
```

File: github/table_github_tree.go [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+	}
+	logger.Error("Nothing found")
```
Comment (bigdatasourav) [inline github/table_github_tree.go:85]:
```suggestion
```

File: github/table_github_tree.go [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+	if err != nil {
+		logger.Error("Error getting tree", err)
+		return nil, err
+	}
+
+	getResp := getResponse.(GetResponse)
+	tree := getResp.tree
+	if tree != nil {
+		logger.Trace("Returning tree", tree)
```
Comment (bigdatasourav) [inline github/table_github_tree.go:82]:
```suggestion
```

File: github/table_github_tree.go [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+				{Name: "tree_sha", Require: plugin.Required},
+				{Name: "recursive", Require: plugin.Optional},
+			},
+			ShouldIgnoreError: isNotFoundError([]string{"404"}),
+			Hydrate:           tableGitHubTreeGet,
+		},
+		Columns: []*plugin.Column{
+			// Top columns
+			{Name: "repository_full_name", Type: proto.ColumnType_STRING, Transform: transform.FromQual("repository_full_name"), Description: "Full name of the repository"},
+			{Name: "tree_sha", Type: proto.ColumnType_STRING, Transform: transform.FromQual("tree_sha"), Description: "Tree SHA"},
```
Comment (bigdatasourav) [inline github/table_github_tree.go:31]:
```suggestion
			{Name: "sha", Type: proto.ColumnType_STRING, Transform: transform.FromQual("sha"), Description: "Tree SHA"},
```

File: github/table_github_tree.go (hunk 3 above)
Comment (bigdatasourav) [inline github/table_github_tree.go:55]:
```suggestion
	sha := quals["sha"].GetStringValue()
```

**Comment by bigdatasourav** [comment]
@asfaltboy Hope you are doing good! 
While reviewing the PR I have suggested some changes. Also, there are merge conflicts which need to resolve.

Additionally, add a doc file (github_tree.md), [example](https://github.com/turbot/steampipe-plugin-github/blob/main/docs/tables/github_branch.md).

Please let us know if you need any help.

**Comment by bigdatasourav** [comment]
Hey @asfaltboy, could you please provide us with an update regarding this PR? Let us know if you need any help.

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**